softioc/extension.c

```diff
@@ -85,33 +72,36 @@ void get_field_offsets(
     while (status == 0)
     {
         const char * field_name = dbGetFieldName(&dbentry);
-        int i;
-        for (i = 0; i < field_count; i ++)
-        {
-            if (strcmp(field_names[i], field_name) == 0)
-            {
-                field_offset[i] = dbentry.pflddes->offset;
-                field_size[i]   = dbentry.pflddes->size;
-                field_type[i]   = dbentry.pflddes->field_type;
-            }
-        }
+        PyObject *ost = Py_BuildValue("iii",
+            dbentry.pflddes->offset,
+            dbentry.pflddes->size,
+            dbentry.pflddes->field_type);
+        PyDict_SetItemString(dict, field_name, ost);
```

Comment:
I think a reference to `ost` is being leaked here.  From https://docs.python.org/3/c-api/dict.html#c.PyDict_SetItemString

>  This function does not steal a reference to val.

Reply:
Nice catch, I'm sure you're right here.